src/Data/Profunctor/Unsafe.hs

```diff
@@ -86,7 +92,11 @@ infixl 8 .#
 -- 'lmap' (f '.' g) ≡ 'lmap' g '.' 'lmap' f
 -- 'rmap' (f '.' g) ≡ 'rmap' f '.' 'rmap' g
 -- @
+#if __GLASGOW_HASKELL__ >= 806
+class (forall a . Functor (p a)) => Profunctor p where
+#else
 class Profunctor p where
+#endif
```

Comment:
minimum definitions becomes `lmap | dimap`, as `rmap` have to be `fmap`, which should be stated as a new law.

Reply:
agreed @phadej